internal/subshell/subshell.go

```diff
@@ -140,6 +142,45 @@ func New(cfg sscommon.Configurable) SubShell {
 	return subs
 }
 
+func availableShells() []SubShell {
+	var shells []SubShell
+	for _, shell := range supportedShells {
+		rcFile, err := shell.RcFile()
+		if err != nil {
+			logging.Error("Could not determine rc file for shell %s: %v", shell.Shell(), err)
+			continue
+		}
+
+		if !fileutils.FileExists(rcFile) {
+			continue
+		}
+
+		shells = append(shells, shell)
+	}
+
+	return shells
+}
+
+func ConfigureAvailableShells(cfg sscommon.Configurable, env map[string]string, identifier sscommon.RcIdentification, isAdmin bool) error {
+	activeShell := New(cfg)
```

Comment:
This feels backwards. You refactored a lot of code such that we don't have to pass subshell around, but are now constructing it ONLY for this function.

If we can't reasonably put this function as a method of SubShell then we should instead pass it as an argument.